boxoffice/models/line_item.py

```diff
@@ -235,36 +236,58 @@ def counts_per_date_per_item(item_collection, user_tz):
     return date_item_counts
 
 
-def sales_by_date(dates, user_tz, item_ids):
+def sales_by_date(sales_datetime, item_ids, timezone):
     """
-    Returns the net sales of line items sold on a date.
-    Accepts a list of dates.
-    ['2016-01-01', '2016-01-02'] => {'2016-01-01': }
+    Takes a timezone aware datetime object, item_ids and returns the sales amount accrued
+    during the day for the given datetime.
     """
     if not item_ids:
         return None
 
-    date_sales = {}
+    start_at = midnight_to_utc(sales_datetime, timezone=timezone)
+    end_at = midnight_to_utc(sales_datetime + datetime.timedelta(days=1), timezone=timezone)
+    sales_on_date = db.session.query('sum').from_statement('''SELECT SUM(final_amount) FROM line_item
+        WHERE status=:status AND ordered_at >= :start_at AND ordered_at < :end_at
+        AND line_item.item_id IN :item_ids
+        ''').params(status=LINE_ITEM_STATUS.CONFIRMED, start_at=start_at, end_at=end_at, item_ids=tuple(item_ids)).first()
+    return sales_on_date[0] if sales_on_date[0] else Decimal(0)
```

Comment:
Use `.scalar()` instead of `.first()` to avoid the `[0]` lookup. http://docs.sqlalchemy.org/en/latest/orm/query.html#sqlalchemy.orm.query.Query.scalar